test/stateStore.test.js

```
@@ -0,0 +1,271 @@
+// import { processRawJsonSirenEntity, stateFactory } from '../state/HypermediaState';
+import { assert }  from '@open-wc/testing';
+import { StateStore } from '../state/stateStore.js';
+
+let stateStore;
+const entityID = 'HREF';
+const token = 'TOKEN';
+const defaultState = {
+	entityID: entityID,
+	token: token,
+};
+
+describe('State Store', () => {
+
+	beforeEach(() => {
+		stateStore = new StateStore();
+	});
+
+	describe('Add state', () => {
+		it('should not add an undefined state', () => {
+			const state = undefined;
+			stateStore.add(state);
+			assert.equal(stateStore._states.size, 0, 'State was added to the store');
+		});
+
+		it('should not add an empty state', () => {
+			const state = {};
+			stateStore.add(state);
+			assert.equal(stateStore._states.size, 0, 'State was added to the store');
+		});
+
+		it('should not add a state without token', () => {
+			const state = {
+				entityID: entityID
+			};
+
+			stateStore.add(state);
```

is there any return value in stateStore.add or exception thrown?
If it is silently failing how consumer will know that state was not added ?

---

There is no return on store.add(). That is a good point , do maps generally have a return on add / set methods? not sure what is being done for error handling outside of the store.

---

the set method for maps will return the object.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/set

So, silent fail it is?